arwen/elrondapi/elrondei.go

```diff
@@ -535,7 +525,13 @@ func storageStore(context unsafe.Pointer, keyOffset int32, keyLength int32, data
 	gasToUse := metering.GasSchedule().ElrondAPICost.StorageStore
 	metering.UseGas(gasToUse)
 
-	return storage.SetStorage(key, data)
+	storageStatus, err := storage.SetStorage(key, data)
+	if err != nil {
+		runtime.SignalUserError(err.Error())
```

Comment:
Use `arwen.WithFault()` here, instead of calling `runtime.SignalUserError()` directly. And `storage.SetStorage()` tends to fail due to a SC programming error (forbidden key accessed), so it's not a user error, but an execution failure instead (which makes `arwen.WithFault()` more appropriate).

In this specific case, the last argument to `arwen.WithFault()`, namely `failExecution`, should still be `runtime.ElrondAPIErrorShouldFailExecution()`, currently configured to `true`, because an error in `storage.SetStorage()` should be unrecoverable and cannot be handled by the SmartContract (unless, later, we allow an "expert" mode where errors are caught and handled by the SC, as discussed some months ago).